### PR TITLE
LPS-115817 Code refactoring

### PR DESCRIPTION
**Pull request by alinedoleron**

### CODE REVIEW

**Comment by alinedoleron** [comment]
ci:test:sf

**Comment by alinedoleron** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/resources/META-INF/resources/js/components/list/List.es.js [hunk 1]
```diff
@@ -21,18 +22,29 @@ import {SidebarContext} from '../sidebar/SidebarContext.es';
...
+	const checkType = (field, type) => {
+		switch (type) {
+			case 'color':
+				return <Color hexColor={field} />;
+			case 'date':
+				return formatDate(field);
+			default:
+				return field;
+		}
+	};
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/resources/META-INF/resources/js/components/list/List.es.js:34]: Good job, @alinedoleron. 😄

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/resources/META-INF/resources/js/components/list/List.es.js [hunk 2]
```diff
@@ -21,18 +22,29 @@ import {SidebarContext} from '../sidebar/SidebarContext.es';
...
+
+	const formatDate = (field) => {
+		const locale = themeDisplay.getLanguageId().split('_', 1).join('');
+
+		return moment(field).locale(locale).format('L');
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/resources/META-INF/resources/js/components/list/List.es.js:39]: @alinedoleron, does `moment` handle internally `undefined` parameters, like `field` and `locale`?

**Comment by natocesarrego** [comment]
Hey @alinedoleron, this new pull looks better! Congratulations! 🙂 

I just have one more question, can you please check it?

Thanks.

**Comment by alinedoleron** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/resources/META-INF/resources/js/components/list/List.es.js (hunk 2 above)
Comment (alinedoleron) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/resources/META-INF/resources/js/components/list/List.es.js:39]: @natocesarrego, I've tested here and when moment receives a undefined value it returns an object like:
`Moment {_isAMomentObject: true, _isUTC: false, _pf: {…}, _locale: Locale, _d: Wed Jul 08 2020 17:26:21 GMT-0300 (Brasilia Standard Time), …}`. So I think it handles it.

**Comment by natocesarrego** [comment]
ci:forward

**Comment by natocesarrego** [comment]
ci:forward

**Pull request closed without merging**